src/rust/engine/fs/store/src/lib.rs

```diff
@@ -69,17 +69,43 @@ pub struct UploadSummary {
   pub upload_wall_time: Duration,
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
-pub enum Source {
-  Local,
-  Remote,
+/// A timespan represented as two floating point representations of seconds
+#[derive(Copy, Clone, Debug, PartialEq, Serialize)]
+pub struct TimeSpan {
```

Comment:
We discussed this question offline

Reply:
To sum up the discussion, we need something that can be serialized, which excludes `std::time::Duration` (intentionally abstract so users don't rely on an implementation) and `time::Timespec` (deprecated and not serde aware).
The issues with `float` are:
* potential loss of precision (academic in the case of span since the zipkin API only cares to the microsecond)
* potential annoyances for testability as they aren't supposed to be used for equality comparison

As an alternative, I could raise a PR to create a very simple crate that replaces `time::Timespec` and fills all our usecases (serialization, ffi friendliness, equality comparison, conversion from protobuf::Timestamp). This would avoid juggling between 4 time representations or so.

On the python side, we could also deal with u64, u32 for secs and nanosecs until the py_zipkin interface, and then convert to float.